### ISSUE TITLE
SDK should support passing iLO RedFish Sessions

### ISSUE THREAD

**Issue by ChrisLynchHPE** [HewlettPackard/ilo-sdk-ruby#27]
iLO 4 and iLO 5 REST API/RedFish interface supports [API sessions](https://hewlettpackard.github.io/ilo-rest-api-docs/ilo5/#creating-and-using-sessions) by using auth tokens.  The SDK should allow for the Auth Token to be passed to `Client.new()`:

```Json
{
    "X-Auth-Token": "{token}"
}
```

as an attribute to the constructor.

**Comment by jsmartt** [comment]
Technically this can be accomplished now, but it's a bit ugly. Here's a workaround until this gets added though:

```ruby
# Create a client with dummy credentials:
client = ILO_SDK::Client.new(
  host: 'https://ilo.example.com',
  user: 'Administrator',
  password: 'dummy123'
)

# Then make rest calls, telling the library to ignore authentication,
# and put in your own header:
client.rest_api(:get, '/redfish/v1/Systems', auth: :none, ‘X-Auth-Token’ => 'my_token')
```

This only works for custom requests though; there's no way to override the auth and headers for the helper methods.

**Comment by ChrisLynchHPE** [comment]
Right.  So if someone wanted to say set virtual media mount URL, they couldn't use the `set_url_boot` helper method.  They would need to use the `rest_api` helper to perform he custom operation themselves.  That's a decent workaround until the iLO SDK can natively support this.
